clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -248,6 +247,30 @@ def _get_pods_info(
                     logger.info(f"Skipping {pod.metadata.name} pod ({pod.status.phase})")
         return pods_by_ip, unschedulable_pods, excluded_pods_by_ip
 
+    def _get_pods_info_with_label(
+        self,
+    ) -> Tuple[Mapping[str, List[KubernetesPod]], List[KubernetesPod], Mapping[str, List[KubernetesPod]],]:
+        pods_by_ip: Mapping[str, List[KubernetesPod]] = defaultdict(list)
+        unschedulable_pods: List[KubernetesPod] = []
+        excluded_pods_by_ip: Mapping[str, List[KubernetesPod]] = defaultdict(list)
+
+        exclude_daemonset_pods = self.pool_config.read_bool(
+            "exclude_daemonset_pods",
+            default=staticconf.read_bool("exclude_daemonset_pods", default=False),
+        )
+        label_selector = f"{self.pool_label_key}={self.pool}"
+
+        for pod in self._core_api.list_pod_for_all_namespaces(label_selector=label_selector).items:
```

Comment:
its a little unfortunate that this is the only line that's changing, yet we're duplicating the rest of the function

how long do we anticipate needing this duplication? if its going to be a while, we should maybe do something like having a single version of this function take a handle to a partial function that we've already given the label selector args to, having a single version of this function take/use **kwargs for the list of args we give list_pod_for_all_namespaces, or having  having a single version of this function take in the list of pods it will work on (and this way, we avoid having duplicated code hanging around for a long time)

Reply:
It won't be a while for us to change the definition back to the correct one @nemacysts I will try to make it more better tho!